### PR TITLE
Create Database based on Schema

### PR DESCRIPTION
**Pull request by Mahdi-Niazi**
### In this milestone we:

- Add medical_histories table
- Add patients table
- Add treatments table
- Add medical_histories_treatments table for many to many relationship
- Add invoices table
- Add invoice_item table

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    name varchar(255),
+    type varchar(255),
+);
+
+CREATE TABLE medical_histories (
+    id serial PRIMARY KEY,
+    addmited_at timestamp,
+    patient_id integer REFERENCES patients(id),
+    status varchar(255),
+);
```
Comment (codepantha) [inline schema_based_on_diagram.sql:20]: - [ ] Please use the `CREATE INDEX` statement to explicitly create indexes on the foreign key columns in your respective tables. These indexes can help improve query performance when you're searching or joining data using these foreign key columns. See examples below 👇 
```SQL
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Created a database based on a diagram 📌 
✔️ Created a join table for the "many-to-many" relationship 📌 
✔️ Good commit history

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    CONSTRAINT fk_invoices
+        Foreign Key (invoice_id) REFERENCES invoices(id),
+);
+
+CREATE INDEX ON medical_histories (patient_id);
+CREATE INDEX ON invoices (medical_history_id);
+
+CREATE INDEX ON invoice_items(treatment_id);
+CREATE INDEX ON invoice_items(invoice_id);
+
```
Comment (mckent05) [inline schema_based_on_diagram.sql:65]: - Good job done so far, but as the previous reviewer mentioned please create an index on all foreign keys for all tables. You are yet to do this for the `medical_history_treatment` table. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Mahdi-Niazi, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You have done well creating your database 👏 👏 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (Abenezer-Tilahun) [reply schema_based_on_diagram.sql:65]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Mahdi-Niazi 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**